third_party/move/move-model/src/builder/module_builder.rs

```diff
@@ -3751,6 +3751,13 @@ impl<'env, 'translator> ModuleBuilder<'env, 'translator> {
             if entry.module_id != self.module_id {
                 continue;
             }
+            // If the function is from a script, its return value must be unit.
+            if self.module_name.is_script() && !entry.result_type.is_unit() {
+                self.parent.error(
+                    &entry.loc,
```

Comment:
Using `name_loc` instead of `loc` might be nicer here for UX here, otherwise, we may end up showing a very large function's source only to say its return type is wrong.

Reply:
done